### PR TITLE
#draft more parrallelism and debugging building devnets in ci

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
## Description
* used proper version of statemine spec for local
* clean up and debug nix tools
* speed up devnets build
* sharing substrate chains nix attrs
* remove not used nix
* better comments on composable chain specs

## Checklist

nix run .#devnet-dali works as before

```
2022-10-06 08:52:55.534  INFO main sc_cli::runner: Composable    
2022-10-06 08:52:55.534  INFO main sc_cli::runner: ✌️  version 2.3.2-dirty    
2022-10-06 08:52:55.534  INFO main sc_cli::runner: ❤️  by Composable Developers, 2020-2022    
2022-10-06 08:52:55.534  INFO main sc_cli::runner: 📋 Chain specification: Local Dali Testnet    
2022-10-06 08:52:55.534  INFO main sc_cli::runner: 🏷  Node name: Alice    
2022-10-06 08:52:55.535  INFO main sc_cli::runner: 👤 Role: AUTHORITY    
2022-10-06 08:52:55.535  INFO main sc_cli::runner: 💾 Database: RocksDb at /tmp/composable-finance/polkadot-launch/dali-dev//alice/chains/dali/db/full    
2022-10-06 08:52:55.535  INFO main sc_cli::runner: ⛓  Native runtime: dali-2401 (dali-3.tx1.au1)    
2022-10-06 08:52:55.550  INFO main composable_node::command: Parachain id: Id(2087)    
2022-10-06 08:52:55.550  INFO main composable_node::command: Parachain Account: 5Ec4AhNu5P23CC2GdrmsC8HYBKiidqhPnfSfRQPaH9swr2b7    
2022-10-06 08:52:55.550  INFO main composable_node::command: Is collating: yes    
2022-10-06 08:53:03.565  INFO main sub-libp2p: [Relaychain] 🏷  Local node identity is: 12D3KooWSTxU8ZWdtV3tkQpttX7GkUSnGPYoHFQWsmKK1iJRzkgq    
2022-10-06 08:53:03.709  INFO main sc_sysinfo: [Relaychain] 💻 Operating system: linux    
2022-10-06 08:53:03.709  INFO main sc_sysinfo: [Relaychain] 💻 CPU architecture: x86_64    
2022-10-06 08:53:03.709  INFO main sc_sysinfo: [Relaychain] 💻 Target environment: gnu    
2022-10-06 08:53:03.709  INFO main sc_sysinfo: [Relaychain] 💻 CPU: 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz    
2022-10-06 08:53:03.709  INFO main sc_sysinfo: [Relaychain] 💻 CPU cores: 8    
2022-10-06 08:53:03.709  INFO main sc_sysinfo: [Relaychain] 💻 Memory: 48181MB    
2022-10-06 08:53:03.709  INFO main sc_sysinfo: [Relaychain] 💻 Kernel: 5.10.102.1-microsoft-standard-WSL2    
2022-10-06 08:53:03.709  INFO main sc_sysinfo: [Relaychain] 💻 Linux distribution: Ubuntu 20.04 LTS    
2022-10-06 08:53:03.709  INFO main sc_sysinfo: [Relaychain] 💻 Virtual machine: yes    
2022-10-06 08:53:03.709  INFO main sc_service::builder: [Relaychain] 📦 Highest known block at #35    
2022-10-06 08:53:03.709  INFO tokio-runtime-worker substrate_prometheus_endpoint: [Relaychain] 〽️ Prometheus exporter started at 127.0.0.1:9616    
2022-10-06 08:53:03.711  INFO                 main sc_rpc_server: [Relaychain] Running JSON-RPC HTTP server: addr=127.0.0.1:9934, allowed origins=Some(["http://localhost:*", "http://127.0.0.1:*", "https://localhost:*", "https://127.0.0.1:*", "https://polkadot.js.org"])    
2022-10-06 08:53:03.715  INFO                 main sc_rpc_server: [Relaychain] Running JSON-RPC WS server: addr=127.0.0.1:43873, allowed origins=Some(["http://localhost:*", "http://127.0.0.1:*", "https://localhost:*", "https://127.0.0.1:*", "https://polkadot.js.org"])    
2022-10-06 08:53:03.718  WARN                 main sc_service::config: [Parachain] Using default protocol ID "sup" because none is configured in the chain specs    
2022-10-06 08:53:03.719  INFO                 main sub-libp2p: [Parachain] 🏷  Local node identity is: 12D3KooWQHPCtz5tcjYjTFcHEWsXwW2P3cXZBrXGTNF67wvMWNUs    
2022-10-06 08:53:03.720  INFO tokio-runtime-worker libp2p_mdns::behaviour::iface: [Relaychain] creating instance on iface 172.29.104.223    
2022-10-06 08:53:03.729  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Relaychain] discovered: 12D3KooWPY7fTngdPcxvHcAgXfPGmYZuWGMzHBgUm3hXJtWhLeHo /ip4/172.29.104.223/tcp/30444    
2022-10-06 08:53:03.729  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Relaychain] discovered: 12D3KooWPY7fTngdPcxvHcAgXfPGmYZuWGMzHBgUm3hXJtWhLeHo /ip4/127.0.0.1/tcp/30444    
2022-10-06 08:53:03.729  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Relaychain] discovered: 12D3KooWPY7fTngdPcxvHcAgXfPGmYZuWGMzHBgUm3hXJtWhLeHo /ip6/::1/tcp/30444    
2022-10-06 08:53:03.729  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Relaychain] discovered: 12D3KooWPY7fTngdPcxvHcAgXfPGmYZuWGMzHBgUm3hXJtWhLeHo /ip6/fe80::215:5dff:fe83:2d05/tcp/30444    
2022-10-06 08:53:03.731  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Relaychain] discovered: 12D3KooWSXPjXZ6bghNz42T1kxyDPKwW8SZB5uoYPVDWfABBGNTw /ip4/172.29.104.223/tcp/30445    
2022-10-06 08:53:03.731  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Relaychain] discovered: 12D3KooWSXPjXZ6bghNz42T1kxyDPKwW8SZB5uoYPVDWfABBGNTw /ip4/127.0.0.1/tcp/30445    
2022-10-06 08:53:03.731  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Relaychain] discovered: 12D3KooWSXPjXZ6bghNz42T1kxyDPKwW8SZB5uoYPVDWfABBGNTw /ip6/::1/tcp/30445    
2022-10-06 08:53:03.731  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Relaychain] discovered: 12D3KooWSXPjXZ6bghNz42T1kxyDPKwW8SZB5uoYPVDWfABBGNTw /ip6/fe80::215:5dff:fe83:2d05/tcp/30445    
2022-10-06 08:53:03.753  INFO                 main sc_sysinfo: [Parachain] 💻 Operating system: linux    
2022-10-06 08:53:03.753  INFO                 main sc_sysinfo: [Parachain] 💻 CPU architecture: x86_64    
2022-10-06 08:53:03.753  INFO                 main sc_sysinfo: [Parachain] 💻 Target environment: gnu    
2022-10-06 08:53:03.753  INFO                 main sc_sysinfo: [Parachain] 💻 CPU: 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz    
2022-10-06 08:53:03.753  INFO                 main sc_sysinfo: [Parachain] 💻 CPU cores: 8    
2022-10-06 08:53:03.753  INFO                 main sc_sysinfo: [Parachain] 💻 Memory: 48181MB    
2022-10-06 08:53:03.753  INFO                 main sc_sysinfo: [Parachain] 💻 Kernel: 5.10.102.1-microsoft-standard-WSL2    
2022-10-06 08:53:03.753  INFO                 main sc_sysinfo: [Parachain] 💻 Linux distribution: Ubuntu 20.04 LTS    
2022-10-06 08:53:03.753  INFO                 main sc_sysinfo: [Parachain] 💻 Virtual machine: yes    
2022-10-06 08:53:03.753  INFO                 main sc_service::builder: [Parachain] 📦 Highest known block at #11    
2022-10-06 08:53:03.753  INFO                 main sc_rpc_server: [Parachain] Running JSON-RPC HTTP server: addr=0.0.0.0:45225, allowed origins=None    
2022-10-06 08:53:03.754  INFO                 main sc_rpc_server: [Parachain] Running JSON-RPC WS server: addr=0.0.0.0:46871, allowed origins=None    
2022-10-06 08:53:03.778  INFO tokio-runtime-worker libp2p_mdns::behaviour::iface: [Parachain] creating instance on iface 172.29.104.223    
2022-10-06 08:53:03.780  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Parachain] discovered: 12D3KooWSXPjXZ6bghNz42T1kxyDPKwW8SZB5uoYPVDWfABBGNTw /ip4/172.29.104.223/tcp/30445    
2022-10-06 08:53:03.780  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Parachain] discovered: 12D3KooWSXPjXZ6bghNz42T1kxyDPKwW8SZB5uoYPVDWfABBGNTw /ip4/127.0.0.1/tcp/30445    
2022-10-06 08:53:03.780  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Parachain] discovered: 12D3KooWSXPjXZ6bghNz42T1kxyDPKwW8SZB5uoYPVDWfABBGNTw /ip6/::1/tcp/30445    
2022-10-06 08:53:03.780  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Parachain] discovered: 12D3KooWSXPjXZ6bghNz42T1kxyDPKwW8SZB5uoYPVDWfABBGNTw /ip6/fe80::215:5dff:fe83:2d05/tcp/30445    
2022-10-06 08:53:03.780  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Relaychain] discovered: 12D3KooWQHPCtz5tcjYjTFcHEWsXwW2P3cXZBrXGTNF67wvMWNUs /ip4/172.29.104.223/tcp/31200    
2022-10-06 08:53:03.780  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Parachain] discovered: 12D3KooWPY7fTngdPcxvHcAgXfPGmYZuWGMzHBgUm3hXJtWhLeHo /ip4/172.29.104.223/tcp/30444    
2022-10-06 08:53:03.780  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Parachain] discovered: 12D3KooWPY7fTngdPcxvHcAgXfPGmYZuWGMzHBgUm3hXJtWhLeHo /ip4/127.0.0.1/tcp/30444    
2022-10-06 08:53:03.780  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Relaychain] discovered: 12D3KooWQHPCtz5tcjYjTFcHEWsXwW2P3cXZBrXGTNF67wvMWNUs /ip4/127.0.0.1/tcp/31200    
2022-10-06 08:53:03.780  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Parachain] discovered: 12D3KooWPY7fTngdPcxvHcAgXfPGmYZuWGMzHBgUm3hXJtWhLeHo /ip6/::1/tcp/30444    
2022-10-06 08:53:03.780  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Relaychain] discovered: 12D3KooWQHPCtz5tcjYjTFcHEWsXwW2P3cXZBrXGTNF67wvMWNUs /ip6/::1/tcp/31200    
2022-10-06 08:53:03.780  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Parachain] discovered: 12D3KooWPY7fTngdPcxvHcAgXfPGmYZuWGMzHBgUm3hXJtWhLeHo /ip6/fe80::215:5dff:fe83:2d05/tcp/30444    
2022-10-06 08:53:03.780  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Relaychain] discovered: 12D3KooWQHPCtz5tcjYjTFcHEWsXwW2P3cXZBrXGTNF67wvMWNUs /ip6/fe80::215:5dff:fe83:2d05/tcp/31200    
2022-10-06 08:53:03.780  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Parachain] discovered: 12D3KooWSTxU8ZWdtV3tkQpttX7GkUSnGPYoHFQWsmKK1iJRzkgq /ip4/172.29.104.223/tcp/30334/ws    
2022-10-06 08:53:03.780  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Parachain] discovered: 12D3KooWSTxU8ZWdtV3tkQpttX7GkUSnGPYoHFQWsmKK1iJRzkgq /ip4/127.0.0.1/tcp/30334/ws    
2022-10-06 08:53:03.780  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Parachain] discovered: 12D3KooWSTxU8ZWdtV3tkQpttX7GkUSnGPYoHFQWsmKK1iJRzkgq /ip6/::1/tcp/30334/ws    
2022-10-06 08:53:03.780  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Parachain] discovered: 12D3KooWSTxU8ZWdtV3tkQpttX7GkUSnGPYoHFQWsmKK1iJRzkgq /ip6/fe80::215:5dff:fe83:2d05/tcp/30334/ws    
2022-10-06 08:53:06.266  INFO tokio-runtime-worker babe: [Relaychain] 👶 New epoch 4 launching at block 0x2163…60b9 (block slot 277507131 >= start slot 277507127).    
2022-10-06 08:53:06.266  INFO tokio-runtime-worker babe: [Relaychain] 👶 Next epoch starts at slot 277507137    
2022-10-06 08:53:06.308  INFO tokio-runtime-worker substrate: [Relaychain] ✨ Imported #36 (0x2163…60b9)    
2022-10-06 08:53:06.317  INFO tokio-runtime-worker cumulus-collator: [Parachain] Starting collation. relay_parent=0x2163af0a234a023ba3e1f7a19f30a5a9073d6ab24a929184f0cf134931e560b9 at=0xf4a97018bfc2d9a56f9f6020b48e004467dc3afb5a7fd17b0f00f784c131ceae
2022-10-06 08:53:06.329  INFO tokio-runtime-worker parachain::approval-voting: [Relaychain] Advanced session window for approvals update=Advanced { prev_window_start: 0, prev_window_end: 3, new_window_start: 0, new_window_end: 4 }
2022-10-06 08:53:08.716  INFO tokio-runtime-worker substrate: [Relaychain] 💤 Idle (2 peers), best: #36 (0x2163…60b9), finalized #32 (0x9059…f305), ⬇ 4.7kiB/s ⬆ 3.5kiB/s    
2022-10-06 08:53:08.754  INFO tokio-runtime-worker substrate: [Parachain] 💤 Idle (0 peers), best: #11 (0xf4a9…ceae), finalized #10 (0x9f35…6e74), ⬇ 2.7kiB/s ⬆ 1.4kiB/s    
2022-10-06 08:53:09.579  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Relaychain] discovered: 12D3KooWCrM2hrif4qFB6mBNMeTxebYzzMRwewP91N1cnQyn2Z6K /ip4/172.29.104.223/tcp/31201    
2022-10-06 08:53:09.579  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Relaychain] discovered: 12D3KooWCrM2hrif4qFB6mBNMeTxebYzzMRwewP91N1cnQyn2Z6K /ip6/::1/tcp/31201    
2022-10-06 08:53:09.579  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Relaychain] discovered: 12D3KooWCrM2hrif4qFB6mBNMeTxebYzzMRwewP91N1cnQyn2Z6K /ip6/fe80::215:5dff:fe83:2d05/tcp/31201    
2022-10-06 08:53:09.579  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Relaychain] discovered: 12D3KooWCrM2hrif4qFB6mBNMeTxebYzzMRwewP91N1cnQyn2Z6K /ip4/127.0.0.1/tcp/31201    
2022-10-06 08:53:09.579  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Parachain] discovered: 12D3KooWCrM2hrif4qFB6mBNMeTxebYzzMRwewP91N1cnQyn2Z6K /ip4/172.29.104.223/tcp/31201    
2022-10-06 08:53:09.579  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Parachain] discovered: 12D3KooWCrM2hrif4qFB6mBNMeTxebYzzMRwewP91N1cnQyn2Z6K /ip6/::1/tcp/31201    
2022-10-06 08:53:09.579  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Parachain] discovered: 12D3KooWCrM2hrif4qFB6mBNMeTxebYzzMRwewP91N1cnQyn2Z6K /ip6/fe80::215:5dff:fe83:2d05/tcp/31201    
2022-10-06 08:53:09.579  INFO tokio-runtime-worker libp2p_mdns::behaviour: [Parachain] discovered: 12D3KooWCrM2hrif4qFB6mBNMeTxebYzzMRwewP91N1cnQyn2Z6K /ip4/127.0.0.1/tcp/31201    
2022-10-06 08:53:12.034  INFO tokio-runtime-worker substrate: [Relaychain] ✨ Imported #37 (0x7236…55a7)    
2022-10-06 08:53:12.038  INFO tokio-runtime-worker cumulus-collator: [Parachain] Starting collation. relay_parent=0x7236e7f2dc5c2b473f600cf560aa2a6dd6d897306bf5fb53025c33eed18055a7 at=0xf4a97018bfc2d9a56f9f6020b48e004467dc3afb5a7fd17b0f00f784c131ceae
2022-10-06 08:53:13.717  INFO tokio-runtime-worker substrate: [Relaychain] 💤 Idle (3 peers), best: #37 (0x7236…55a7), finalized #33 (0x167c…18d7), ⬇ 3.6kiB/s ⬆ 3.2kiB/s    
2022-10-06 08:53:13.755  INFO tokio-runtime-worker substrate: [Parachain] 💤 Idle (1 peers), best: #11 (0xf4a9…ceae), finalized #10 (0x9f35…6e74), ⬇ 1.1kiB/s ⬆ 1.0kiB/s    

```

### CODE REVIEW

**Review by alexkdefi — APPROVED** [review]
👍

File: .nix/polkadot-bin.nix [hunk 1]
```diff
@@ -1,32 +1,18 @@
...
-    hash = "sha256-LEz3OrVgdFTCnVwzU8C6GeEougaOl2qo7jS9qIdMqAM=";
+    # until we update to 0.9.28 cannot update to this revision (needed to run on rococo-local relay)
+    #rev = "v0.9.27";
+    rev = "6882eff7961008a54749fa8ed445e40844febc3a";
+    #hash = "sha256-LEz3OrVgdFTCnVwzU8C6GeEougaOl2qo7jS9qIdMqAM=";
+    hash = "sha256-Fv/auvHytPr/bF4SGK/Cwl7PvWml+ZSchkE7GGPYcaU=";
   };
-
-  cargoSha256 = "sha256-6y+WK2k1rhqMxMjEJhzJ26WDMKZjXQ+q3ca2hbbeLvA=";
+  #cargoHash = "sha256-eSZGz7HXjLxQ584OMTVQNBOWPVFBtGWQfuJid489Xm4=";
```
Comment (andor0) [inline .nix/polkadot-bin.nix:15]: did you forget to remove it?

File: flake.nix [hunk 2]
```diff
@@ -608,23 +608,26 @@
                 docker volume prune -f
               '';
 
+            system-prune = pkgs.writeShellScriptBin "composable-system-prune" ''
+              ls ${docker-wipe-system}/bin/docker-wipe-system
+              rm /tmp/composable-finance/polkadot-launch/ -r -f && rm *.log -r -f
```
Comment (andor0) [inline flake.nix:613]:
```suggestion
              rm -rf /tmp/composable-finance/polkadot-launch  *.log
```

**Review by andor0 — APPROVED** [review]

**Review by KaiserKarel — APPROVED** [review]

**Comment by dzmitry-lahoda** [comment]
Closing this in favor of more nixified CI/CD instead of manual attempts to build DAG.

File: flake.nix (hunk 2 above)
Comment (dzmitry-lahoda) [reply flake.nix:613]: obsolete, because will use zombienet and it generates randoms

**Pull request closed without merging**